src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -3050,6 +3066,12 @@ void addProperties(ImmutableArray<ParameterSymbol> recordParameters)
                         builder.InstanceInitializersForRecordDeclarationWithParameters.Insert(addedCount, new FieldOrPropertyInitializer.Builder(property.BackingField, paramList.Parameters[param.Ordinal]));
                         addedCount++;
                     }
+                    else if (existingMember.Kind switch { SymbolKind.Property => true, SymbolKind.Field => true, _ => false })
+                    {
```

Comment:
>{ [](start = 20, length = 1)

What if this existing member is `static`? The required code to be generated for `Deconstruct` would be erroneous. So it must be an error.

Reply:
Fixed.

Reply:
Since we're checking for an explicit `GetMethod`, consider testing with base property with `set` only.